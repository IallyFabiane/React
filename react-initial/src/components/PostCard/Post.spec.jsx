import { toBeInTheDocument } from "@testing-library/jest-dom/dist/matchers";
import { render, screen } from "@testing-library/react";
import { PostCard } from '.';

const mockProps = {
    title: 'title1',
    body: 'body1',
    id: 1,
    cover: 'img/img.png'
}

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard {...mockProps} />);

    expect(screen.getByRole('img', { name: 'title1' })).toHaveAttribute('src', mockProps.cover)
    expect(screen.getByRole('heading', { name: 'title1' })). toBeInTheDocument()
    expect(screen.getByText('body1')). toBeInTheDocument()
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard {...mockProps} />)
        expect(container.firstChild).toMatchSnapshot()

    });
})