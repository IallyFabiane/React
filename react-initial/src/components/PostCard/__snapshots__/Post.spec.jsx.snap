// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      title1
    </h1>
    <p>
      body1
    </p>
  </div>
</div>
`;
